Python/Product/Analysis/Values/TypingTypeInfo.cs

```diff
@@ -0,0 +1,329 @@
+                    if (args.Count == 0) {
+                        return null;
+                    }
+                    return Scope.GetOrMakeNodeValue(_node, NodeValueKind.None, n => {
+                        var p = new ProtocolInfo(_unit.ProjectEntry);
+                        p.AddReference(n, _unit);
+                        var callArgs = GetTypeList(args[0]) ?? new[] { args[0] };
+                        p.AddProtocol(new CallableProtocol(
+                            p,
+                            null,
+                            callArgs,
+                            ToInstance(args.ElementAtOrDefault(1) ?? AnalysisSet.Empty)
+                        ));
+                        return p;
+                    });
+
+
+                case "Iterable":
+                case "Iterator": {
+                        var iter = Scope.GetOrMakeNodeValue(_node, NodeValueKind.Iterator, n => {
+                            var p = new ProtocolInfo(_unit.ProjectEntry);
+                            p.AddReference(n, _unit);
+                            p.AddProtocol(new IterableProtocol(p, AnalysisSet.UnionAll(args.Select(ToInstance))));
+                            return p;
+                        });
+                        return (name == "Iterator") ? iter.GetIterator(_node, _unit) : iter;
+                    }
+
+                case "Generator":
+                    return Scope.GetOrMakeNodeValue(_node, NodeValueKind.Iterator, n => {
+                        var gi = new GeneratorInfo(State, Entry);
+                        if (args.Count >= 1) {
+                            gi.AddYield(n, _unit, ToInstance(args[0]), false);
+                        }
+                        if (args.Count >= 2) {
+                            gi.AddSend(n, _unit, ToInstance(args[1]), false);
+                        }
+                        if (args.Count >= 3) {
+                            gi.AddReturn(n, _unit, ToInstance(args[2]), false);
+                        }
+                        return gi;
+                    });
+                case "NamedTuple": return null;
+            }
+
+            return null;
+        }
+
+        public IAnalysisSet Finalize(string name) {
+            if (string.IsNullOrEmpty(name)) {
+                return null;
+            }
+
+            switch (name) {
+                case "Callable": return ClassInfo[BuiltinTypeId.Function];
+                case "Tuple": return ClassInfo[BuiltinTypeId.Tuple];
+                case "Container": return ClassInfo[BuiltinTypeId.List];
+                case "ItemsView": return ClassInfo[BuiltinTypeId.DictItems];
+                case "Iterable": {
+                        var p = new ProtocolInfo(Entry);
+                        p.AddReference(_node, _unit);
+                        p.AddProtocol(new IterableProtocol(p, AnalysisSet.Empty));
+                        return p;
+                    }
+                case "Iterator": {
```

Comment:
>Iterator [](start = 22, length = 8)

case looks identical to above, merge?

Reply:
We'd switch between `IterableProtocol` and `IteratorProtocol` then, but otherwise they are the same. I can merge - it isn't that ugly.